### PR TITLE
[BOOTDATA] Improve 'Open With' registry information

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

This PR will improve 'Open With' registry information.
JIRA issue: [CORE-14677](https://jira.reactos.org/browse/CORE-14677)

## Proposed changes

- Update boot/bootdata/hivecls.inf and boot/bootdata/hivesft.inf .

Before:
open-with-before

After:
open-with-after

### CODE REVIEW

File: boot/bootdata/hivesft.inf [hunk 1]
```diff
@@ -3,8 +3,103 @@ Signature="$ReactOS$"
 
 [AddReg]
 
-HKLM,"SOFTWARE\Classes\Applications\iexplore.exe","",0x00000000,""
-HKLM,"SOFTWARE\Classes\Applications\iexplore.exe\shell\open\command","",0x00000000,"""%programfiles%\Internet Explorer\iexplore.exe"" %1"
+; Applications
+HKLM,"SOFTWARE\Classes\Applications\notepad.exe\shell\open\command","",0x00020000,"%SystemRoot%\system32\notepad.exe %1"
+HKLM,"SOFTWARE\Classes\Applications\mspaint.exe\shell\open\command","",0x00020000,"%SystemRoot%\system32\mspaint.exe %1"
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply boot/bootdata/hivesft.inf]: Why not ""%1""? This is not escape sequence.

[the remark this answers is not in the document]
Comment (binarymaster) [reply boot/bootdata/hivesft.inf]: I've looked at wrong line 453... :sweat_smile: 
```inf
HKCR,"SHCmdFile\shell\open\ddeexec","",0x00000000,"[ShellFile(\"%1\",\"%1\",%S)]"
```
Fixed now.

File: boot/bootdata/hivesft.inf [hunk 2]
```diff
@@ -3,26 +3,121 @@ Signature="$ReactOS$"
...
+HKLM,"SOFTWARE\Classes\Applications\wordpad.exe\shell\open\command","",0x00020000,"%SystemRoot%\system32\wordpad.exe %1"
```
Comment (HBelusca) [inline boot/bootdata/hivesft.inf]: wordpad also needs the file parameter in `""%1""`

File: boot/bootdata/hivesft.inf [hunk 3]
```diff
@@ -3,26 +3,121 @@ Signature="$ReactOS$"
...
+HKLM,"SOFTWARE\Classes\Applications\iexplore.exe\shell\open\command","",0x00020000,"""%programfiles%\Internet Explorer\iexplore.exe"" %1"
```
Comment (HBelusca) [inline boot/bootdata/hivesft.inf]: Ditto (since you've added `""%1""` for other iexplore.exe lines below).

File: boot/bootdata/hivesft.inf (hunk 2 above)
Comment (katahiromz) [reply boot/bootdata/hivesft.inf]: OK.

File: boot/bootdata/hivesft.inf (hunk 3 above)
Comment (katahiromz) [reply boot/bootdata/hivesft.inf]: OK.

**Comment by katahiromz** [comment]
Done.

File: boot/bootdata/hivecls.inf [hunk 4]
```diff
@@ -24,7 +32,7 @@ HKCR,".clp","Content Type",0x00000000,"application/x-msclip"
 HKCR,"clpfile","",0x00000000,"Clipboard Element"
 HKCR,"clpfile","FriendlyTypeName",0x00020000,"@%SystemRoot%\system32\clipbrd.exe,-121"
 HKCR,"clpfile\DefaultIcon","",0x00020000,"%SystemRoot%\system32\clipbrd.exe,1"
-HKCR,"clpfile\shell\open\command","",0x00020000,"%SystemRoot%\system32\clipbrd.exe %1"
+HKCR,"clpfile\shell\open\command","",0x00020000,"%SystemRoot%\system32\clipbrd.exe ""%1"""
```
Comment (HBelusca) [inline boot/bootdata/hivecls.inf]: Not sure about this one; have you tested something like:
```
clipbrd.exe "C:\Documents and settings\Administrator\Desktop\test.clp"
```
?

File: boot/bootdata/hivecls.inf [hunk 5]
```diff
@@ -241,15 +249,15 @@ HKCR,".ico","PerceivedType",0x00000000,"image"
 HKCR,"icofile","",0x00000000,"Icon"
 HKCR,"icofile","FriendlyTypeName",0x00020000,"@%SystemRoot%\system32\shell32.dll,-213"
 HKCR,"icofile\DefaultIcon","",0x00000000,"%1"
-HKCR,"icofile\shell\open\command","",0x00020000,"rundll32.exe %SystemRoot%\system32\shimgvw.dll,ImageView_Fullscreen %1"
+HKCR,"icofile\shell\open\command","",0x00020000,"rundll32.exe %SystemRoot%\system32\shimgvw.dll,ImageView_Fullscreen ""%1"""
```
Comment (HBelusca) [inline boot/bootdata/hivecls.inf]: Does it work better with the quotes?

Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: In Windows 10: 
```
HKEY_CLASSES_ROOT\Applications\photoviewer.dll\shell\print\command=%SystemRoot%\System32\rundll32.exe "%ProgramFiles%\Windows Photo Viewer\PhotoViewer.dll", ImageView_Fullscreen %1
```
So, Windows doesn't needs double quotes for %1.

I think Windows does workarround on interpretation about %1.

File: boot/bootdata/hivecls.inf (hunk 4 above)
Comment (katahiromz) [reply boot/bootdata/hivecls.inf]:
```c
int main(void)
{
printf("%s\n", GetCommandLineA());
return 0;
}
```

File: boot/bootdata/hivecls.inf (hunk 5 above)
Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: In Windows, mspaint.exe doesn't do this workarround.

Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: base/system/rundll32/rundll32.c (383):
```c
    // The rest of the arguments will be passed to dll function
    if (i < argc)
        lptCmdLine = argv[i];
    else
        lptCmdLine = _T("");
```

rundll32 passes the rest commandline, so it doesn't require to add quotes.

Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: https://support.microsoft.com/en-us/help/164787/info-windows-rundll-and-rundll32-interface

> Note in particular that this means a DLL in the "C:\Program Files" folder should be converted to its short name.
> The dllname may not contain any spaces or commas or quotation marks. This is a limitation in the Rundll command line parser.

File: boot/bootdata/hivecls.inf (hunk 4 above)
Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: clip-ignore
Oh, no! Win2k3 ClipBook Viewer is simply ignoring the parameters. What should we do?

File: boot/bootdata/hivecls.inf (hunk 5 above)
Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: The problem is whether adding quotes affects 16-bit compatibility.

Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: A Win16 app can access the registry tree. He doesn't assume it was quoted.

File: boot/bootdata/hivecls.inf (hunk 4 above)
Comment (HBelusca) [reply boot/bootdata/hivecls.inf]: On Win2k3, you can save (using the clipbrd) the clipboard contents to a file (with extension .clp).
Then try opening that file when it's inside a long-named directory with spaces.
Try also to do the same, but using ReactOS' clipbrd.exe.

File: boot/bootdata/hivecls.inf (hunk 5 above)
Comment (HBelusca) [reply boot/bootdata/hivecls.inf]: Also, as indicated in the URL you provided, everything after the name of the function in the rundll32 command line is passed to the function (here, "ImageView_Fullscreen"), so you should not touch (ie. add quotes) around it.

File: boot/bootdata/hivecls.inf (hunk 4 above)
Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: clipbrd1clipbrd2
Both cases are failed in both of text and bitmap formats.

File: boot/bootdata/hivecls.inf (hunk 5 above)
Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: Reverted.

File: boot/bootdata/hivecls.inf (hunk 4 above)
Comment (katahiromz) [reply boot/bootdata/hivecls.inf]: I think the clipbrd program should accept both of quoted and unquoted file paths.

**Comment by katahiromz** [comment]
Let's go!

**Pull request merged**